docs/dockerfile_experimental.md

```diff
@@ -0,0 +1,128 @@
+# Dockerfile frontend experimental syntaxes
+
+The contents of this document is planned to be moved to https://docs.docker.com in future.
+
+## Note for Docker users
+
+If you are using Docker v18.06 or later, BuildKit mode can be enabled by setting `export DOCKER_BUILDKIT=1` on the client side.
+Docker v18.06 also requires the daemon to be running as the experimental mode.
+
+You need to use `docker build` CLI instead of `buildctl` CLI mentioned in this document.
+See [the `docker build` document](https://docs.docker.com/engine/reference/commandline/build/) for the usage.
+
+Note that not all upstream BuildKit features are available in Docker BuildKit-mode.
+
+## Use experimental Dockerfile frontend
+The features mentioned in this document are experimentally available as [`tonistiigi/dockerfile:master-experimental`](https://hub.docker.com/r/tonistiigi/dockerfile/tags/) image.
+
+To use the experimental features, the first line of your Dockerfile needs to be `# syntax=tonistiigi/dockerfile:master-experimental`.
+As the experimental syntaxes may change in future revisions, you may want to pin the image to a specific revision.
+
+NOTE: `tonistiigi/dockerfile` is planned to be renamed to `docker/dockerfile-upstream` soon. See [#528](https://github.com/moby/buildkit/issues/528).
+
+## Experimental syntaxes
+
+### `RUN --mount=type=bind` (the default mount type)
+
+This mount type allows binding directories (read-only) in the context or in an image to the build container.
+
+|Option               |Description|
+|---------------------|-----------|
+|`target`(required)   | Mount path.|
+|`source`             | Source path in the `from`. Defaults to the root of the `from`.|
+|`from`               | Image name. Defaults to the build context.|
+
+
+### `RUN --mount=type=cache`
+
+This mount type allows the build container to cache directories for compilers and package managers.
+
+|Option               |Description|
+|---------------------|-----------|
+|`id`                 | ID of the cache.|
+|`target`(required)   | Mount path.|
+|`ro`,`readonly`      | Read-only if set.|
+|`sharing`            | Either one of `shared`, `private`, or `locked`. Defaults to `shared`. `shared` cache mount can be used concurrently by multiple writers. `private` creates a new mount if there are multiple writers. `locked` pauses second writer until first one releases the mount.|
```

Comment:
remove "Either"

Reply:
And I think:
Either "`shared` cache mount*s* can..." or "*A* `shared cache mount can..."
and:
 "`locked` pauses *the* second ...."